app/static/app/js/ModelView.jsx

```diff
@@ -0,0 +1,1067 @@
+            depthMaterial.uniforms.octreeSize.value = pointcloud.pcoGeometry.boundingBox.size().x;
+            depthMaterial.bbSize = pointcloud.material.bbSize;
+            depthMaterial.treeType = pointcloud.material.treeType;
+            
+            scenePointCloud.overrideMaterial = depthMaterial;
+            renderer.clearTarget( rtDepth, true, true, true );
+            renderer.render(scenePointCloud, camera, rtDepth);
+            scenePointCloud.overrideMaterial = null;
+          }
+          
+          {// ATTRIBUTE PASS
+            attributeMaterial.size = pointSize;
+            attributeMaterial.pointSizeType = pointSizeType;
+            attributeMaterial.screenWidth = width;
+            attributeMaterial.screenHeight = height;
+            attributeMaterial.pointColorType = pointColorType;
+            attributeMaterial.depthMap = rtDepth;
+            attributeMaterial.uniforms.visibleNodes.value = pointcloud.material.visibleNodesTexture;
+            attributeMaterial.uniforms.octreeSize.value = pointcloud.pcoGeometry.boundingBox.size().x;
+            attributeMaterial.fov = camera.fov * (Math.PI / 180);
+            attributeMaterial.spacing = pointcloud.pcoGeometry.spacing;
+            attributeMaterial.near = camera.near;
+            attributeMaterial.far = camera.far;
+            attributeMaterial.heightMin = heightMin;
+            attributeMaterial.heightMax = heightMax;
+            attributeMaterial.intensityMin = pointcloud.material.intensityMin;
+            attributeMaterial.intensityMax = pointcloud.material.intensityMax;
+            attributeMaterial.setClipBoxes(pointcloud.material.clipBoxes);
+            attributeMaterial.clipMode = pointcloud.material.clipMode;
+            attributeMaterial.bbSize = pointcloud.material.bbSize;
+            attributeMaterial.treeType = pointcloud.material.treeType;
+            
+            scenePointCloud.overrideMaterial = attributeMaterial;
+            renderer.clearTarget( rtNormalize, true, true, true );
+            renderer.render(scenePointCloud, camera, rtNormalize);
+            scenePointCloud.overrideMaterial = null;
+          }
+          
+          {// NORMALIZATION PASS
+            normalizationMaterial.uniforms.depthMap.value = rtDepth;
+            normalizationMaterial.uniforms.texture.value = rtNormalize;
+            Potree.utils.screenPass.render(renderer, normalizationMaterial);
+          }
+          
+          pointcloud.material = originalMaterial;
+            
+          volumeTool.render();
+          renderer.clearDepth();
+          profileTool.render();
+          measuringTool.render();
+          transformationTool.render();
+        }
+
+
+      }
+    };
+
+    function loop() {
+      requestAnimationFrame(loop);
+      update();
+      
+      if(quality === "Splats"){
+        if(!highQualityRenderer){
+          highQualityRenderer = new HighQualityRenderer();
+        }
+        highQualityRenderer.render(renderer);
```

Comment:
It gets initialized only if you select the "splats" display for points. I moved the reference to the `highQualityRenderer` closer to `loop` so hopefully it's more clear.